tests/src/python/providertestbase.py

```diff
@@ -127,6 +127,22 @@ def testRectAndExpression(self):
         expected = [4]
         assert set(expected) == result, 'Expected {} and got {} when testing for combination of filterRect and expression'.format(set(expected), result)
 
+    def testGetFeaturesLimit(self):
+        it = self.provider.getFeatures(QgsFeatureRequest().setFeatureLimit(2))
+        features = [f['pk'] for f in it]
+        assert len(features) == 2, 'Expected two features, got {} instead'.format(features)
```

Comment:
Do you want to print `len(features)` in the error message?

Also, consider using `self.assertEqual(len(features), 2)` which gives a pretty decent, though generic error message, out of the box. The benefit being that you don't have to worry about composing a (maybe buggy) error message.